src/components/CommitGraph/index.tsx

```diff
@@ -120,6 +122,7 @@ export default function CommitGraph({
                 dateFormatFn={dateFormatFn}
                 currentBranch={currentBranch}
                 fullSha={fullSha}
+                onClick={() => onClick && onClick(commits[index])}
```

Comment:
we don't need `index` here, `commit` is already passed, should remove the index change above.
also, avoid creating a function when `onClick` is not defined.

```suggestion
               onClick={onClick ? () => onClick(commit) : undefined}
```